test/src/receiver/main.cpp

```diff
@@ -1,129 +1,165 @@
+
+#include "../../../include/receiver_cadmium.hpp"
+
+using namespace std;
+
+using hclock = chrono::high_resolution_clock;
+using TIME = NDTime;
+
+
+/***** SETING INPUT PORTS FOR COUPLEDs *****/
+struct inp : public cadmium::in_port<Message_t> {};
+
+/***** SETING OUTPUT PORTS FOR COUPLEDs *****/
+struct outp : public cadmium::out_port<Message_t> {};
+
+
+/********************************************/
+/****** APPLICATION GENERATOR *******************/
+/********************************************/
+template<typename T>
+class ApplicationGen : public iestream_input<Message_t,T> {
+    public:
+        ApplicationGen() = default;
+        ApplicationGen(const char* file_path) :
+                iestream_input<Message_t,T>(file_path) {
+    }
+};
+
+
+int main() {
+    auto start = hclock::now(); //to measure simulation execution time
+
+    /*************** Loggers *******************/
+    static std::ofstream out_data(
+        "../test/data/receiver/receiver_test_output.txt");
+    struct oss_sink_provider {
+        static std::ostream& sink() {
+            return out_data;
+        }
+    };
+
+    using
+        info = cadmium::logger::logger<cadmium::logger::logger_info,
+            cadmium::dynamic::logger::formatter<TIME>,
+                oss_sink_provider>;
+    using
+        debug = cadmium::logger::logger<cadmium::logger::logger_debug,
+            cadmium::dynamic::logger::formatter<TIME>,
+                oss_sink_provider>;
+    using
+        state = cadmium::logger::logger<cadmium::logger::logger_state,
+            cadmium::dynamic::logger::formatter<TIME>,
+                oss_sink_provider>;
+    using
+        log_messages = cadmium::logger::logger<cadmium::logger::logger_messages,
+            cadmium::dynamic::logger::formatter<TIME>,
+                oss_sink_provider>;
+    using
+        routing = cadmium::logger::logger<cadmium::logger::logger_message_routing,
+            cadmium::dynamic::logger::formatter<TIME>,
+                oss_sink_provider>;
+    using
+        global_time = cadmium::logger::logger<cadmium::logger::logger_global_time,
+            cadmium::dynamic::logger::formatter<TIME>,
+                oss_sink_provider>;
+    using
+        local_time = cadmium::logger::logger<cadmium::logger::logger_local_time,
+            cadmium::dynamic::logger::formatter<TIME>,
+                oss_sink_provider>;
+    using
+        log_all = cadmium::logger::multilogger <info,
+            debug, state, log_messages, routing, global_time, local_time>;
+
+    using logger_top = cadmium::logger::multilogger<log_messages, global_time>;
+
+
+    /*******************************************/
+
+
+
+    /********************************************/
+    /****** APPLICATION GENERATOR *******************/
+    /********************************************/
+    string input_data_control =
+        "../test/data/receiver/receiver_input_test.txt";
+    const char * i_input_data_control = input_data_control.c_str();

```

Comment:
there wont be any space betwenn char and *

Reply:
solved